src/mono/wasm/host/BrowserHost.cs

```diff
@@ -88,29 +89,104 @@ private async Task RunAsync(ILoggerFactory loggerFactory, CancellationToken toke
         await host.WaitForShutdownAsync(token);
     }
 
-    private async Task<(ServerURLs, IWebHost)> StartWebServerAsync(string appPath, bool forwardConsole, string[] urls, CancellationToken token)
+    private async Task<(ServerURLs, IWebHost)> StartWebServerAsync(BrowserArguments args, string[] urls, CancellationToken token)
     {
-        WasmTestMessagesProcessor? logProcessor = null;
-        if (forwardConsole)
+        Func<WebSocket, Task>? onConsoleConnected = null;
+        if (args.ForwardConsoleOutput ?? false)
         {
-            logProcessor = new(_logger);
+            WasmTestMessagesProcessor logProcessor = new(_logger);
+            onConsoleConnected = socket => RunConsoleMessagesPump(socket, logProcessor!, token);
         }
 
-        WebServerOptions options = new
-        (
-            OnConsoleConnected: forwardConsole
-                                    ? socket => RunConsoleMessagesPump(socket, logProcessor!, token)
-                                    : null,
-            ContentRootPath: Path.GetFullPath(appPath),
-            WebServerUseCors: true,
-            WebServerUseCrossOriginPolicy: true,
-            Urls: urls
-        );
-
-        (ServerURLs serverURLs, IWebHost host) = await WebServer.StartAsync(options, _logger, token);
+        // If we are using new browser template, use dev server
+        if (TryCreateDevServerOptions(args, urls, onConsoleConnected, out var devServerOptions))
+            return await DevServer.DevServer.StartAsync(devServerOptions, _logger, token);
+
+        // Otherwise for old template, use web server
+        WebServerOptions webServerOptions = CreateWebServerOptions(urls, args.CommonConfig.AppPath, onConsoleConnected);
+        (ServerURLs serverURLs, IWebHost host) = await WebServer.StartAsync(webServerOptions, _logger, token);
```

Comment:
It is kinda unclear on why/how we would decide between devserver and the webserver. The code reads as "if creating devserver options" fails for any reason then we fall back to webserver, which would not be correct.

Instead, I would suggest extracting the code from `TryCreateDevServerOptions` to find the static webassets manifest to a separate method, and wrap that in a `IsProjectUsingWasmSdk` (or whatever is relevant). Then the above code can be straight forward to use `DevServer` if the project is using WasmSdk, and WebServer if it is the old style.

Reply:
Resolved by specifying additional parameter, as suggested below